packages/salesforcedx-vscode-soql/src/editor/soqlUtils.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ModelDeserializer, ModelSerializer } from '@salesforce/soql-model';
+import { Impl, Soql, SoqlModelUtils } from '@salesforce/soql-model/lib';
```

Comment:
For readability I'd recommend a more specific name than `Impl`, or at least in this file (by using import as).

Reply:
I'm undecided a bit here as the soql-model is written that way, and exports those members...I guess I'm used to seeing that.  @jgellin-sf - what's your thoughts?

Reply:
Additional to that, you shouldn't have to import using `/lib`, you can change the library's package.json so that everything you need to consume can be imported by just using `@salesforce/soql-model`

Reply:
Naming is hard. I like the interfaces being defined separately from the implementations because that makes things in the model really flexible independent of implementation. There are a couple of conventions for implementations of interfaces, non-prefixed interfaces and Impl-postfixed implementations which is what I went with, or I-prefixed interfaces and non-prefixed implementations, which seems like an anti-convention in typescript for some reason. Grouping the various interfaces and the various implementations with namespaces felt right as a way to make it clear what we are dealing with in certain situations, but the namespaces themselves are maybe not named that well. 'Soql' for the SOQL model namespace is maybe ok, but 'Impl' for the model implementation namespace lacks specificity. 

@xyc I am onboard with the `import { Impl as SoqlModelImpl }` or some such renaming for readability, but maybe it's better to rename the entire namespace at the source. Out of curiosity, what conventions do you favor for interface-implementation naming?

Reply:
Regarding @lcampos comments, the soql-model lib is set up as he describes, so all of those imports can be in one statement.

Reply:
> Out of curiosity, what conventions do you favor for interface-implementation naming
I don't think we have specific naming to separate interface vs implementation, but `SoqlModelImpl` seems fine in this case

Reply:
We are going to change this at the source.  and then we will update this.